### PR TITLE
added password reseting functionality using token and added email fun…

### PR DESCRIPTION
**Pull request by vaibhav-d-simformsolutions**
…ctionality using nodemailer package

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (sumitsimform) [inline .vscode/settings.json:1]: add .vscode folder in .gitignore

File: auth-middleware/validations.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
+const joi = require("joi");
+joi.objectId = require("joi-objectid")(joi);
+
+exports.registerSchema = joi.object({
+  email: joi
+    .string()
+    .lowercase()
+    .email()
+    .required()
+    .error(() => new Error("Invalid Email !")),
```
Comment (sumitsimform) [inline auth-middleware/validations.js:10]: Add case : - string contains @

File: auth-middleware/validations.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  password: joi
+    .string()
+    .min(4)
+    .required()
+    .error(() => new Error("Invalid Password !")),
+  confirmPassword: joi.ref("password"),
+  _csrf: joi
```
Comment (sumitsimform) [inline auth-middleware/validations.js:17]: why we have csrf as _csrf 
don't use 2 formats in single project use only csrf.

**Review by sumitsimform — CHANGES_REQUESTED** [review]
Resolve all this comments

File: .vscode/settings.json (hunk 1 above)
Comment (vaibhav-d-simformsolutions) [reply .vscode/settings.json:1]: added

**Pull request merged**